### DISCUSSION TITLE
A change about SpamBlocker API integration

### DISCUSSION THREAD

**Discussion by aj3423** [haumacher/phoneblock#117] [General]
Hi, I'm finalizing the last piece and encountered some problems.

Previously, we discussed reporting a number when it's blocked by either an API query or the local Spam DB which was originally identified by an API query. I've been trying to implement it like: report it as the same category as in the query result. For example, if the query returns "E_ADVERTISING", report it as "E_ADVERTISING". 

However, I found it's quite complex to implement, it requires an extra layer between the querying and reporting, which would significantly complicate the code, and make it bug-prone. It will be even more complex if some API use different values for querying and reporting, etc.

Therefore, I decide to simplify it by reporting them as "B_MISSED". In short, all numbers that are reported automatically will have the type "B_MISSED", only manual reporting will have different types.

Are you okay with that?

**Comment by haumacher** [comment]
Yes, B_MISSED für all automatic reports is the best thing you can do.

**Reply by aj3423** [reply]
Thanks.
